apps/ta_dump.py

```diff
@@ -113,9 +116,132 @@ def adc_integral_hist(adc_integrals):
+
+    plt.title("TA Time Windows Summary")
+    plt.xlabel("Ticks")
+
+    plt.tight_layout()
+    plt.savefig("ta_time_windows_summary.svg")
+    plt.close()
+
+def plot_summary_stats(ta_data, no_anomaly=False, quiet=False):
+    """
+    Plot summary statistics on various TA member data.
+    Displays as box plots on multiple pages of a PDF.
+    """
+    # 'Sanity' titles _should_ all be the same value.
+    titles = {
+                'adc_integral': "ADC Integral Summary",
+                'adc_peak': "ADC Peak Summary",
+                'algorithm': "Algorithm (Sanity) Summary",
+                'channel_end': "Channel End Summary",
+                'channel_peak': "Channel Peak Summary",
+                'channel_start': "Channel Start Summary",
+                'detid': "Detector ID (Sanity) Summary",
+                'num_tps': "Number of TPs Summary",
+                'time_end': "Time End Summary",
+                'time_peak': "Time Peak Summary",
+                'time_start': "Time Start Summary",
+                'type': "Type (Sanity) Summary"
+             }
+    labels = {
+                'adc_integral': "ADC Integral",
+                'adc_peak': "ADC Count",
+                'algorithm': "",
+                'channel_end': "Channel Number",
+                'channel_peak': "Channel Number",
+                'channel_start': "Channel Number",
+                'detid': "",
+                'num_tps': "TP Count",
+                'time_end': "Ticks",
+                'time_peak': "Ticks",
+                'time_start': "Ticks",
+                'type': ""
+             }
+
+    anomaly_filename = 'ta_anomaly_summary.txt'
+
+    if not no_anomaly:
+        if not quiet:
+            print(f"Writing descriptive statistics to {anomaly_filename}.")
+        import os
+        from scipy import stats
+        if os.path.isfile(anomaly_filename):
+            # Prepare a new ta_anomaly_summary.txt
+            os.remove(anomaly_filename)
+
+    with PdfPages("ta_summary_stats.pdf") as pdf:
+        for ta_key, title in titles.items():
+            # Extract only ta_key
+            plot_data = []
+            for frag_data in ta_data:
+                for ta in frag_data:
+                    plot_data = plot_data + list(ta[ta_key])
+            plot_data = np.array(plot_data)
+
+            plt.figure(figsize=(6,4))
+            ax = plt.gca()
+
+            plt.boxplot(plot_data, notch=True, vert=False, sym='+')
+            plt.yticks([]) # Would just show '1'.
+            ax.xaxis.grid(True)
+            plt.xlabel(labels[ta_key])
+
+            plt.title(title)
+
+            plt.tight_layout()
+            pdf.savefig()
+            plt.close()
+
+            # Write anomalies to file.
+            if not no_anomaly:
+                if "Sanity" in title and np.all(plot_data == plot_data[0]):
+                    # Either passed check or all wrong in the same way.
+                    continue
```

Comment:
```suggestion
            if no_anomaly:
                continue
                    
            if "Sanity" in title and np.all(plot_data == plot_data[0]):
                # Either passed check or all wrong in the same way.
                 continue
```
With that, the lines after that don't need to be indented.

Reply:
I understand that this makes the indentation better, but is having this repeated `if "Sanity" ...` better?

Reply:
I think it would still make it more readable & explicit, ofc not without 0 downsites. Could also have one longer if statement with brackets, e.g. `if (not no_anomaly) and "Sanity" in title` etc